docs/DeveloperGuide.md

```diff
@@ -1,34 +1,283 @@
 # Developer Guide
 
-## Design & Implementation
+## 1. Setting Up
+   * Prerequisites
+     * JDK 11 or above
+     * IntelliJ IDE
+   * Setting up the project in your computer
+     * Fork this repo, and clone the fork to your computer
+     * Open IntelliJ (if you are not in the welcome screen, click File > Close Project to close the existing project dialog first)
+     * Set up the correct JDK version for Gradle
+     * Click Configure > Project Defaults > Project Structure
+     * Click New…​ and find the directory of the JDK
+     * Click Import Project
+     * Locate the build.gradle file and select it. Click OK
+     * Click Open as Project
+     * Click OK to accept the default settings.
+   * Verifying the setup
+     * Run the jikan.jikan and try a few commands
+     * Run the tests and ensure they all pass.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 2. Design
+![image_info](./pictures/ClassDiagram.png)
```

Comment:
To add on to the above, it is not only not a requirement, but also not recommended by the professor to state all attributes and methods for a class diagram. You are recommended to include important details that will benefit the reader when they read your development guide. 

It will also be clearer to include a brief explanation for you Fig 2.1. Class diagram of the Jikan program,  to guide the reader through your UML diagram.

Reply:
TA here.

General rule of thumb: More important for the diagram to be comprehensible than comprehensive.

@damithc